File: 646-maximum-length-of-pair-chain/maximum-length-of-pair-chain.java
class Solution {
    public int findLongestChain(int[][] pairs) {
        Arrays.sort(pairs,new Comparator<int[]>(){
            @Override
            public int compare(int[] a,int[] b){
                return Integer.compare(a[1],b[1]);
            }
        });
        int n=pairs.length,ans=1;
        int[] dp=new int[n];
        dp[0]=1;
        for(int i=1;i<n;i++){
            for(int j=i-1;j>=0;j--){
                if(pairs[j][1]<pairs[i][0]) dp[i]=Math.max(dp[j]+1,dp[i]);
            }
            ans=Math.max(ans,dp[i]);
        }
        return ans;
    }
}

File: 646-maximum-length-of-pair-chain/maximum-length-of-pair-chain.cpp
class Solution {
public:
static bool comp(vector<int>& a,vector<int>& b){
    return a[1]<b[1];
}
    int findLongestChain(vector<vector<int>>& pairs) {
        sort(pairs.begin(),pairs.end(),comp);
        int ans=0;
        // for(auto a:pairs) cout<<a[0]<<" "<<a[1];
        for(int i=0;i<pairs.size();i++){
            int cure=pairs[i][1];
            int cur=1;
            for(int j=i+1;j<pairs.size();j++){
                if(cure<pairs[j][0]){
                    cur++;
                    cure=pairs[j][1];
                }
            }
            if(cur>ans) ans=cur;
        }
        return ans;
    }
};